spec/controllers/verify_controller_spec.rb

```diff
@@ -16,53 +17,95 @@
       }
     end
     let(:profile) { subject.idv_session.profile }
+    let(:step) { 'personal_key' }
 
-    subject(:response) { get :show }
+    subject(:response) { get :show, params: { step: step } }
 
     before do
-      stub_sign_in
+      allow(IdentityConfig.store).to receive(:idv_api_enabled_steps).
+        and_return(idv_api_enabled_steps)
+      stub_sign_in(user)
       stub_idv_session
     end
 
-    context 'with step feature-disabled' do
-      before do
-        allow(IdentityConfig.store).to receive(:idv_api_enabled_steps).and_return([])
-      end
+    it 'renders 404' do
+      expect(response).to be_not_found
+    end
+
+    context 'with idv api enabled' do
+      let(:idv_api_enabled_steps) { ['something'] }
 
-      it 'renders 404' do
-        expect(response).to be_not_found
+      context 'invalid step' do
+        let(:step) { 'bad' }
+
+        it 'renders 404' do
+          expect(response).to be_not_found
+        end
       end
-    end
 
-    context 'with step feature-enabled' do
-      before do
-        allow(IdentityConfig.store).to receive(:idv_api_enabled_steps).
-          and_return(['personal_key', 'personal_key_confirm'])
+      context 'with personal key step enabled' do
+        let(:idv_api_enabled_steps) { ['personal_key', 'personal_key_confirm'] }
+
+        before do
+          profile_maker = Idv::ProfileMaker.new(
+            applicant: applicant,
+            user: user,
+            user_password: password,
+          )
+          profile = profile_maker.save_profile
+          controller.idv_session.pii = profile_maker.pii_attributes
+          controller.idv_session.profile_id = profile.id
+          controller.idv_session.personal_key = profile.personal_key
+        end
+
+        it 'renders view' do
+          expect(response).to render_template(:show)
+        end
+
+        it 'sets app data' do
+          response
+
+          expect(assigns[:app_data]).to include(
+            app_name: APP_NAME,
+            base_path: idv_app_path,
+            completion_url: idv_gpo_verify_url,
```

Comment:
We might need to rethink how we handle "completion URL", since it can be dynamic based on how the user proceeds through the flow. For example, if they opt to confirm address by mailed letter.

Couple thoughts:

- Move this handling into a controller whose route can be statically populated into the application, and whose action logic would determine where to redirect user
- Change Personal Key step to submit to an endpoint which would be expected to respond with the redirect URL